sycl/source/device_selector.cpp

```diff
@@ -21,11 +22,17 @@ namespace sycl {
 // Utility function to check if device is of the preferred backend.
 // Currently preference is given to the level0 backend.
 static bool isDeviceOfPreferredSyclBe(const device &Device) {
```

Comment:
```suggestion
static bool isLevelZeroGPUDevice(const device &Device) {
```
New code below using this function with current generic name is very confusing. It seems like it should work with any device type and old code applied this function to all device type looks reasonable. Now it's applied only to GPU device type and it's not obvious why.

Reply:
@bader  This function is used to just give a bonus point (+50) when deciding which device to choose when multiple devices are available with the same score.
So, it is mainly used for GPU devices for now.
This function will give a bonus point if the device "Device" is matched with SYCL_BE. Not only for Level0. So your naming suggestion is not quite right.

Reply:
`isPreferredGPUDevice`?

Reply:
That makes more sense. :)